js/app/modules/card/knowledgeDocument.js

```diff
@@ -237,7 +247,7 @@ const KnowledgeDocument = new Module.Class({
     _get_webview: function () {
         let webview = this._create_webview();
 
-        webview.renderer.enable_scroll_manager = this.show_toc;
+        webview.renderer.enable_scroll_manager = true;
```

Comment:
Oh. Maybe the answer is to check in the decision handler if the scroll manager is enabled. If it is not, then return false and let webkit's default decision handler navigate to `#anchor`?